Create an English translation of "(Nunca) Ouvi Falar de Rust"

ErichDonGubler opened the pull request:
This PR is the culmination of work started at https://github.com/ErichDonGubler/bltavares-presentations/issues/1. Happy to contribute a translation to your awesome Rust presentation, Bruno!

## Remaining work

- [x] Initial translation pass
- [x] Editing pass
    - [x] Handle any TODO/FIXME/XXX comments that Erich can do himself
- [ ] Questions/requests for Bruno
    - [x] Resolution for FIXME/TODO items
    - [x] Code overflows on a screen @erichdongubler tested: https://i.imgur.com/KJB3ZgA.png
        - There are several places this happens. Style problem? :(
    - [x] Some `rustc` errors overflow on display: https://i.imgur.com/m9d9a7S.png
        - There are several places this happens too. :(
    - [x] Localize Nickel walkthrough assets?
    - [x] Anything else you'd like in the translation?

ErichDonGubler committed:
Create an English translation of "(Nunca) Ouvi Falar de Rust"

bltavares committed:
Updates the title of the document to English

bltavares committed:
Generates the presentation as HTML from the english source

bltavares commented:
Hey @ErichDonGubler, thank you so much for the effort translating it. This last weekend I wasn't feelin so well, so sorry for the delay on coming back to it. It means a lot to me that you liked the presentation.

I've generated the HTML for a quick preview of how it is looking like, and I'll go through the TODOs over this weekend. I'll keep you posted.

Meanwhile, I've updated the branch with the HTML. Have a nice day!

ErichDonGubler committed:
Whoops, finish translating the final slides of (Nunca) Ouvi Falar de Rust and fix up some errors/missed sentences -- update both the ORG and HTML representations

ErichDonGubler commented:
@bltavares: My pleasure! Jamais quero deixar de falar português nem de dar de volta o carinho que tenho recebido do povo brasileiro. :)

Also, just added some more tweaks. It was nice to see the presentation in its entirety -- it let me find a few more QA-ish things I added to the issues in the OP. :)

ErichDonGubler commented:
@bltavares: My pleasure! Jamais quero deixar de falar português nem de dar de volta o carinho que tenho recebido do povo brasileiro. :) Espero que teja tudo melhorzinho já!

Also, just added some more tweaks. It was nice to see the presentation in its entirety -- it let me find a few more QA-ish things I added to the issues in the OP. :)

bltavares committed:
Revisiting the presentation to fix TODOs and small adjustments

There were some images that I didn't put back on the source .org mode that I've
added right before presenting. This commit adds them as well on the translation.

bltavares committed:
Adds the images on the source pt-br file

bltavares committed:
Make Rust a highlighted word on the presentation

bltavares committed:
Make large code blocksk fit on screen

bltavares committed:
Removes partial highlight from Rustup

bltavares committed:
Introduces highlights on the presentation slides

bltavares committed:
Cross link the English presentation on the pt-br presentation

bltavares commented:
@ErichDonGubler Happy new year! Sorry for the delay, the last couple of days were complicated to get back to this (lack of internet, holidays, getting sick and so on).

I've updated the slides with some color and made some small changes. I was thinking of merging it to have it available as a link already, and further changes we make as a new PR. What do you think?

Would you mind checking if the changes I've made read ok in English as well?

Thank you for the work put on it (:

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.html:
```diff
@@ -223,14 +227,14 @@ <h2 id="org26774a6">"Why did they choose to create a low-level language?"</h2>
 <section >
 <p>
-Memory management without a garbage collector
+Memory management without a Garbage Collector
```
Capitalized? Any special reason for this?

bltavares replied:
I'm not sure, I felt like this is a named concept, so we should capitalize it? Would it be better to make it lower cased?

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.html:
```diff
@@ -170,13 +175,12 @@ <h2 id="rust"><a id="org53ddb70"></a>Rust</h2>
-
 <p>
-What stood out to me for continued learning:
+What kept me interested on the language:
```
AH, this is what you meant! Aprende-se algo novo a cada dia. :)

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -300,81 +300,81 @@ error[E0282]: type annotations needed
-      }
+   #+BEGIN_SRC rust
+     #[derive(Debug)]
+     struct Order {
```
Ahhh, também faz sentido isso. Perdão por ter presumido que quis dizer "request"! :)

bltavares replied:
No problem at all! It is a word with double mapping to English (:

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -455,7 +461,7 @@ $ cargo doc --open
 
    #+reveal: split
-   And a great option to construct low-level platforms
+   And a great option to construct high-level platforms
```
...derp.

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -485,9 +491,9 @@ $ cargo doc --open
 
 ** A website that's:
-   - Light
+   - Lightweight
```
Oooh, this is a nice mapping to s'more English techno-jargon!

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -840,41 +847,47 @@ $ cargo run
-   - There's [[https://riot.im/app/#/room/#mozilla_#rust-br:matrix.org][#rust-br]] for Portuguese speakers
-   - The [[https://telegram.me/rustlangbr][@rustlangbr]] group on Telegram
+   - There is a [[https://doc.rust-lang.org/book/][free book]] on the website [EN]
+   - [[https://riot.im/app/#/room/#mozilla_#rust-beginners:matrix.org][#rust-beginners]] is a great channel to help with initial doubts
```
"doubts" is not usually the term native American English speakers use for problems one may run into while learning something -- I'm assuming that what you're saying here is in that vein. I'd definitely use "questions" here.

bltavares replied:
Thank you, I didn't know about this nuance. I'll change it.

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -840,41 +847,47 @@ $ cargo run
-   - [[http://exercism.io/languages/rust/about][Exercism]]
-   - And [[https://github.com/rust-br/estudos][online study group]] in Portuguese
+   - [[http://exercism.io/languages/rust/about][Coding exercises]]
+   - And an [[https://github.com/rust-br/estudos][online studygroup]] in Portuguese
```
"studygroup" isn't a word in English -- is there a reason you removed the space here?

bltavares replied:
I'll fix it back hehe

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.en.org:
```diff
@@ -840,41 +847,47 @@ $ cargo run
 
+** Special thanks
+
+   [[https://github.com/ErichDonGubler][Erich Gubler]]
```
:D

ErichDonGubler reviewed (commented):
I'm glad you could clarify some things I wasn't sure about. This is good for me as a translator! I'm glad that this is useful to you. :) I know revision is a lot easier than feeling like you have a montinho of translation work to do...

ErichDonGubler commented on nunca-ouvi-falar-de-rust/index.org:
```diff
@@ -9,6 +9,10 @@
 
+  #+reveal: split
+  There are slides in [[https://presentations.bltavares.com/nunca-ouvi-falar-de-rust/index.en.html%0A][English]]
+  Thanks to [[https://github.com/ErichDonGubler][Erich Gubler]]
```
:D

ErichDonGubler commented:
I think that getting this out faster and tweaking later is great! I'd love to see this live. :)

bltavares committed:
More fix suggestions from Erich

bltavares commented:
So it has been merged! Thank you for helping with the translation and sorry for how long it took it all. I hope it wasn't too much of a bad experience.

The presentation should be online at some point after Github's and Cloudflare's cache clears up here https://presentations.bltavares.com/nunca-ouvi-falar-de-rust/index.en.html
